src/Arduino_GigaDisplayTouch.h

```diff
@@ -52,87 +54,123 @@ typedef struct GDTpoint_s GDTpoint_t;
-       * @brief Attach an interrupt handler function for touch detection callbacks.
-       * @param handler The pointer to the user-defined handler function.
-       */
-      void onDetect(void (*handler)(uint8_t, GDTpoint_t*));
-  private:
-      TwoWire&          _wire;
-      uint8_t           _intPin;
-      mbed::InterruptIn _irqInt; 
-      uint8_t           _rstPin;
-      uint8_t           _addr;
-      GDTpoint_t        _points[GT911_MAX_CONTACTS];
-      void              (*_gt911TouchHandler)(uint8_t, GDTpoint_t*);
-
-      uint8_t   _gt911WriteOp(uint16_t reg, uint8_t data);
-      uint8_t   _gt911WriteBytesOp(uint16_t reg, uint8_t * data, uint8_t len);
-      uint8_t   _gt911ReadOp(uint16_t reg, uint8_t * data, uint8_t len);
-      void      _gt911onIrq();
-      uint8_t   _gt911ReadInputCoord(uint8_t * pointsbuf, uint8_t& contacts);
+public:
+#ifdef __MBED__
+/**
+ * @brief Construct a new touch controller for Giga Display Shield.
+ *
+ * @param wire A reference to the Wire interface to be used for communication
+ * with the touch controller.
+ * @param intPin The interrupt pin number for the touch controller.
+ * @param rstPin The reset pin number for the touch controller.
+ * @param addr The device address for the touch controller.
+ */
+#if defined(ARDUINO_GIGA)
+  Arduino_GigaDisplayTouch(TwoWire &wire = Wire1,
+                           uint8_t intPin = PinNameToIndex(PI_1),
+                           uint8_t rstPin = PinNameToIndex(PI_2),
+                           uint8_t addr = GT911_I2C_ADDR_BA_BB);
+#elif defined(ARDUINO_PORTENTA_H7_M7)
+  Arduino_GigaDisplayTouch(TwoWire &wire = Wire,
+                           uint8_t intPin = PinNameToIndex(PD_4),
+                           uint8_t rstPin = PinNameToIndex(PD_5),
+                           uint8_t addr = GT911_I2C_ADDR_BA_BB);
+#else
+  Arduino_GigaDisplayTouch(TwoWire &wire, uint8_t intPin, uint8_t rstPin,
+                           uint8_t addr);
+#endif
+#elif defined(__ZEPHYR__)
+  Arduino_GigaDisplayTouch();
+#endif
+
+  ~Arduino_GigaDisplayTouch();
+
+  /**
+   * @brief Initialize the touch controller.
+   *
+   * @return true If the touch controller is successfully initialized, false
+   * Otherwise
+   */
+  bool begin();
+
+  /**
+   * @brief De-initialize the touch controller.
+   */
+  void end();
+
+#if defined(__MBED__)
+  /**
+   * @brief Check if a touch event is detected and get the touch points.
+   * @param points The array containing the coordinates of the touch points.
+   * @return uint8_t The number of detected touch points.
+   */
+  uint8_t getTouchPoints(GDTpoint_t *points);
+#elif defined(__ZEPHYR__)
+  /**
+   * @brief Check if a touch event is detected and get the touch points.
+   * @param points The array containing the coordinates of the touch points.
+   * @param timeout in microseconds.
+   * @return uint8_t The number of detected touch points.
+   */
+  uint8_t getTouchPoints(GDTpoint_t *points, uint32_t timeout = 0);
```

Comment:
The API should be the same for both. 
```suggestion
  /**
   * @brief Check if a touch event is detected and get the touch points.
   * @param points The array containing the coordinates of the touch points.
   * @return uint8_t The number of detected touch points.
   */
  uint8_t getTouchPoints(GDTpoint_t *points);
```